### DISCUSSION TITLE
Cypress and WinAppDriver (or any other desktop app testing tool)

### DISCUSSION THREAD

**Discussion by diogogmaio** [cypress-io/cypress#27247] [Questions and Help]
I am looking to recommend Cypress as the tool for Integration, E2E and Regression testing for a project but one key part is the ability to test an office addin (specifically Word). I know that it can't do that type of testing.
But is there any (tested) combination of tools that I can securely reccomend? or a library, plugin connected with Cypress?

So for example Appium can work with WinAppDriver. Could be Cypress + Appium a good fit?

I am kinda all over the place with so many tools and possibilities but in the end i still like CYpress.
What are your recommendations?
Someone told me to forget Cypress and go with CodeceptJS and all the integrations but i am not so sure.

**Comment by portyu9** [comment]
Hi @diogogmaio 

I would keep these as **separate test layers** rather than trying to make Cypress control Word through WinAppDriver.

Cypress is a browser automation framework; it cannot drive the native Microsoft Word window/UI. An Office add-in complicates things because the add-in itself is web technology, but `Office.js` is initialized inside an Office-hosted webview, not inside the normal browser Cypress launches.

A practical split is:

1. **Cypress** for the web application/UI portions that can run in a normal browser.
2. **Unit tests** for your add-in business logic, with the Office APIs behind an abstraction. Microsoft currently provides `Office-Addin-Mock` specifically for mocking `Word`, `Office`, etc.
3. A **separate Windows UI automation suite** for the smaller number of scenarios that genuinely need to open desktop Word and verify the full add-in/host integration. WinAppDriver/Appium or another Windows UI automation tool belongs at this layer.

I would run those suites independently in CI and combine their results rather than trying to expose WinAppDriver as Cypress commands.

If Word on the web can cover some of your real host integration scenarios, that may let you keep more coverage browser-based, but desktop Word itself is outside Cypress's control.

So I would still recommend Cypress if most of the product is web-based, but I would not present Cypress as a replacement for the native Office-host testing part of your existing suite.

The full discussion was reviewed; there were no attached screenshots or replies. Current Cypress, Microsoft Office testing, and WinAppDriver documentation were also checked.

Please mark this answer as accepted if it helped, thank you.
